### DISCUSSION TITLE
Add documentation

### DISCUSSION THREAD

**Discussion by ruststarter** [Byron/google-apis-rs#473] [Ideas]
The documentation is weak, in particular about ID's.
For example, books1 mylibrary bookshelves-add-volume requires a bookshelf id and a volume id.
It does not say what these id's are. One can figure out the bookshelf id is obtained from bookshelves list. But I have yet no clue about what is a volume id.
I have tried the ISBN id, it does not work. 
How do I add a volume by giving its ISBN number ?

**Comment by Byron** [comment]
The documentation is generated from the json-schema that is available for each API. If they use the same ID for many different kinds of objects without further describing what ID that is supposed to be in the method documentation, there is nothing to go on.

In these situations you'd probably want to head over to the official Google docs for the respective API and see if that sheds some light.

*If it interests you*, a great deal of time was spent in making the documentation what it is today, and it could be much, much worse.

**Reply by ruststarter** [reply]
Hi,
I agree that the problem is not the documentation of your CLI. It is more a problem of tutorial for Google books. I did not find in Google docs API a clear explanation of their own volume id, or how you can get the volume id from the ISBN with a query. I found it browsing on the net.

https://www.googleapis.com/books/v1/volumes?q=isbn:<isbn>

Moreover, there are issues with this query for some translated book. Sometimes this query does not find the google id, for example for ISBN 9782743607395. However, if you query the french books on with the same ISBN using another query
https://books.google.fr/books?vid=ISBN<isbn>

then it works and you find the volume id kkQjGQAACAAJ . Why ? beats me...

It would be worth some explanations somewhere but is on Google side, not the CLI. Now I also have a problem however with the CLI.

I have several books whose volume id start with character '-', for example -WfJQgAACAAJ for ISBN 9782258020986

It is interpreted by CLI as an argument. I have quoted but it does not work. I tried

books1 mylibrary bookshelves-add-volume 1001 "-WfJQgAACAAJ" 

=> error: Found argument '-W' which wasn't expected, or isn't valid in this context

also tried
books1 mylibrary bookshelves-add-volume 1001 '-WfJQgAACAAJ'
=> error: Found argument '-W' which wasn't expected, or isn't valid in this context

and

books1 mylibrary bookshelves-add-volume 1001 \-WfJQgAACAAJ
error: Found argument '-W' which wasn't expected, or isn't valid in this context

How to enter a volume id starting with '-' ?

**Reply by Byron** [reply]
Thanks for sharing, maybe somebody reading this can help.

Otherwise, please do not expect any support for Google APIs here, this is just a repository containing a lot of generated code. Maybe the generated CLI can do it, maybe it can't, they are not individually tested and everything was finished years ago.
